subsys/bluetooth/services/fast_pair/CMakeLists.txt

```diff
@@ -51,41 +51,69 @@ if(NOT SYSBUILD)
     )
 
-  set(FP_PROVISIONING_DATA_ADDRESS $<TARGET_PROPERTY:partition_manager,PM_BT_FAST_PAIR_ADDRESS>)
-
   if(NOT DEFINED FP_MODEL_ID OR NOT DEFINED FP_ANTI_SPOOFING_KEY)
     message(FATAL_ERROR "FP_MODEL_ID and FP_ANTI_SPOOFING_KEY build variables must be specified.")
   endif()
 
-  add_custom_command(
-    OUTPUT
-    ${FP_PROVISIONING_DATA_HEX}
-    DEPENDS
-    "${CMAKE_BINARY_DIR}/pm.config"
-    COMMAND
-    ${PYTHON_EXECUTABLE} ${ZEPHYR_NRF_MODULE_DIR}/scripts/nrf_provision/fast_pair/fp_provision_cli.py
-                         -o ${FP_PROVISIONING_DATA_HEX} -a ${FP_PROVISIONING_DATA_ADDRESS}
-                         -m ${FP_MODEL_ID} -k ${FP_ANTI_SPOOFING_KEY}
-    COMMENT
-    "Generating Fast Pair provisioning data hex file"
-    USES_TERMINAL
-    )
+  if(CONFIG_PARTITION_MANAGER_ENABLED)
+    set(FP_PROVISIONING_DATA_ADDRESS $<TARGET_PROPERTY:partition_manager,PM_BT_FAST_PAIR_ADDRESS>)
+
+    add_custom_command(
+      OUTPUT
+      ${FP_PROVISIONING_DATA_HEX}
+      DEPENDS
+      "${CMAKE_BINARY_DIR}/pm.config"
+      COMMAND
+      ${PYTHON_EXECUTABLE} ${ZEPHYR_NRF_MODULE_DIR}/scripts/nrf_provision/fast_pair/fp_provision_cli.py
+                          -o ${FP_PROVISIONING_DATA_HEX} -a ${FP_PROVISIONING_DATA_ADDRESS}
+                          -m ${FP_MODEL_ID} -k ${FP_ANTI_SPOOFING_KEY}
+      COMMENT
+      "Generating Fast Pair provisioning data hex file"
+      USES_TERMINAL
+      )
 
-  add_custom_target(
-    ${FP_PARTITION_NAME}_target
-    DEPENDS
-    "${FP_PROVISIONING_DATA_HEX}"
-    )
+    add_custom_target(
+      ${FP_PARTITION_NAME}_target
+      DEPENDS
+      "${FP_PROVISIONING_DATA_HEX}"
+      )
 
-  set_property(
-    GLOBAL PROPERTY
-    ${FP_PARTITION_NAME}_PM_HEX_FILE
-    "${FP_PROVISIONING_DATA_HEX}"
-    )
+    set_property(
+      GLOBAL PROPERTY
+      ${FP_PARTITION_NAME}_PM_HEX_FILE
+      "${FP_PROVISIONING_DATA_HEX}"
+      )
 
-  set_property(
-    GLOBAL PROPERTY
-    ${FP_PARTITION_NAME}_PM_TARGET
-    ${FP_PARTITION_NAME}_target
-    )
+    set_property(
+      GLOBAL PROPERTY
+      ${FP_PARTITION_NAME}_PM_TARGET
+      ${FP_PARTITION_NAME}_target
+      )
+  else()
+    dt_nodelabel(bt_fast_pair_partition_node NODELABEL "bt_fast_pair_partition")
+    dt_reg_addr(bt_fast_pair_partition_addr PATH ${bt_fast_pair_partition_node})
+    set(FP_PROVISIONING_DATA_ADDRESS ${bt_fast_pair_partition_addr})
+
+    execute_process(
+      COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_NRF_MODULE_DIR}/scripts/nrf_provision/fast_pair/fp_provision_cli.py
+                          -o ${FP_PROVISIONING_DATA_HEX} -a ${FP_PROVISIONING_DATA_ADDRESS}
+                          -m ${FP_MODEL_ID} -k ${FP_ANTI_SPOOFING_KEY}
+      COMMAND_ECHO STDOUT
+      )
+
+    set_property(
+      GLOBAL APPEND PROPERTY
+      HEX_FILES_TO_MERGE
+      ${FP_PROVISIONING_DATA_HEX}
+      ${CMAKE_BINARY_DIR}/zephyr/zephyr.hex
+      )
```

Comment:
I don't really have time at present to go through this in detail or check it but at first glance this does not seem right, it assumes that the output is names zephyr.hex, which is not the case. The output filename is set with a Kconfig, and there's also the possibility of using MCUboot signing which generates a signed image

Reply:
> The output filename is set with a Kconfig

Alright, I was not aware of the `CONFIG_KERNEL_BIN_NAME` Kconfig option. Thanks, fixed.

> there's also the possibility of using MCUboot signing which generates a signed image

I think that MCUboot forces to use Partition Manager so should not be a problem here as this path is only valid for the `CONFIG_PARTITION_MANAGER_ENABLED=n`